optuna/integration/keras.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import absolute_import
+
+import math
+
+import optuna
+
+from typing import Dict  # NOQA
+
+try:
+    from keras.callbacks import Callback
+    _available = True
+except ImportError as e:
+    _import_error = e
+    # KerasPruningExtension is disabled because Keras is not available.
+    _available = False
+    # This alias is required to avoid ImportError at KerasPruningExtension definition.
+    Callback = object
+
+
+class KerasPruningCallback(Callback):
+    """Keras callback to prune unpromising trials.
+
+    Example:
+
+        Add a pruning callback which observes validation losses.
+
+        .. code::
+
+            model.fit(X, y,
+                callbacks=KerasPruningCallback(trial, 'val/loss'))
+
+    Args:
+        trial:
+            A :class:`~optuna.trial.Trial` corresponding to the current evaluation of the
+            objective function.
+        observation_key:
+            An evaluation metric for pruning, e.g., ``val/loss`` and
+            ``val/acc``. Please refer to
+            `keras.Model reference <https://keras.io/models/about-keras-models/>`_ for further details. 
+    """  # NOQA
```

Comment:
I think this `NOQA` can be removed if you insert an appropriate newline to the above line.

And please execute the `$ autopep8 optuna/integration/keras.py --in-place` command to fix some style problems that cannot be detected by `flake8`.